core/src/main/java/com/linecorp/armeria/server/logging/structured/util/ServiceLogJsonFormat.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured.util;
+
+import java.io.IOException;
+import java.util.function.Consumer;
+
+import org.apache.thrift.TApplicationException;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TMessage;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.transport.TMemoryBuffer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import com.linecorp.armeria.common.thrift.ApacheThriftCall;
+import com.linecorp.armeria.common.thrift.ApacheThriftReply;
+import com.linecorp.armeria.common.thrift.text.TTextProtocol;
+
+/**
+ * A utility to provide JSON based service log serialization.
+ * The JSON serialization provided by this class deal with some known classes in a special way.
+ * For the representation classes of Apache Thrift RPC objects such as {@link TBase} and
+ * {@link TApplicationException}, it uses {@link TTextProtocol} to serialize objects in a human-readable format.
+ */
+public final class ServiceLogJsonFormat {
+    private ServiceLogJsonFormat() {}
+
+    /**
+     * Returns newly created {@link ObjectMapper} which is configured properly to serialize some knows classes
+     * in a good way.
+     */
+    public static ObjectMapper objectMapper(SimpleModule... userModules) {
```

Comment:
letting users to apply further customization is my intention. i.e, when user want to add more serializers for their original classes but still want to inherit custom serializers for Thrift objects?